packages/near-membrane-base/src/environment.ts

```diff
@@ -1,192 +1,181 @@
-                    const blueDistortedGetter = this.distortionCallback(
-                        blueGetter
-                    ) as () => BlueValue;
-                    currentBlueGetter = function currentDistortedBlueGetter() {
-                        const value: BlueValue = ReflectApply(
-                            blueDistortedGetter,
-                            broker.getBlueValue(this),
-                            emptyArray
-                        );
-                        return broker.getRedValue(value);
-                    };
-                    redDescriptor.get = function get(): RedValue {
-                        return ReflectApply(currentBlueGetter, this, emptyArray);
-                    };
-                }
+            const configurable = ObjectHasOwnProperty.call(blueDescriptor, 'configurable')
+                ? !!blueDescriptor.configurable
+                : undefinedSymbol;
+            const enumerable = ObjectHasOwnProperty.call(blueDescriptor, 'enumerable')
+                ? !!blueDescriptor.enumerable
+                : undefinedSymbol;
+            const writable = ObjectHasOwnProperty.call(blueDescriptor, 'writable')
+                ? !!blueDescriptor.writable
+                : undefinedSymbol;
+            const valuePointer = ObjectHasOwnProperty.call(blueDescriptor, 'value')
+                ? this.blueGetTransferableValue(blueDescriptor.value)
+                : undefinedSymbol;
+            const getPointer = ObjectHasOwnProperty.call(blueDescriptor, 'get')
+                ? this.blueGetTransferableValue(blueDescriptor.get)
+                : undefinedSymbol;
+            const setPointer = ObjectHasOwnProperty.call(blueDescriptor, 'set')
+                ? this.blueGetTransferableValue(blueDescriptor.set)
+                : undefinedSymbol;
+            // installing descriptor into the red side
+            this.redCallableDefineProperty(
+                oPointer,
+                key,
+                configurable,
+                enumerable,
+                writable,
+                valuePointer,
+                getPointer,
+                setPointer
+            );
+        }
+    }
 
-                if (typeof blueDescriptor.set === 'function') {
-                    redDescriptor.set = function set(v: RedValue): void {
-                        // if a global setter is invoke, the value will be use as it
-                        // is as the result of the getter operation
-                        currentBlueGetter = () => v;
-                    };
+    lazyRemap(o: ProxyTarget, keys: PropertyKey[]) {
+        const enumerablePropertyKeys = Object.keys(o); // except symbols
+        const keysTuple: (PropertyKey | boolean)[] = [];
+        for (let i = 0, len = keys.length; i < len; i += 1) {
+            const key = keys[i];
+            let isEnumerable = true;
+            if (typeof key === 'symbol') {
+                if (ObjectPropertyIsEnumerable.call(o, key)) {
```

Comment:
We should avoid `.call` here ☝️. Maybe make it a generic function like `IsEnumerable`.

Reply:
We should use `ReflectApply` ☝️